mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -444,31 +444,21 @@ class LinearRegressionModel private[ml] (
     @Since("1.4.0") override val uid: String,
     @Since("2.0.0") val coefficients: Vector,
     @Since("1.3.0") val intercept: Double)
-  extends RegressionModel[Vector, LinearRegressionModel]
-  with LinearRegressionParams with MLWritable {
-
-  private var trainingSummary: Option[LinearRegressionTrainingSummary] = None
+  extends RegressionModel[Vector, LinearRegressionModel] with MLWritable
+  with LinearRegressionParams with HasTrainingSummary[LinearRegressionTrainingSummary] {
 
   override val numFeatures: Int = coefficients.size
 
+  private[ml]
+  override def setSummary(summary: Option[LinearRegressionTrainingSummary]): this.type =
```

Comment:
Yeah, why do we need these definitions that just invoke the superclass method?

Reply:
removed.